### PR TITLE
Update Sonar scanner to JDK 11 & fix Coverity and Sonar issues

### PR DESCRIPTION
**Pull request by JirkaAichler**
All problem fixed: https://scan.coverity.com/projects/zowe-sample-spring-boot-api-service?tab=overview

### CODE REVIEW

File: jarpatcher/src/main/java/jarpatcher/JarPatcher.java [hunk 1]
```diff
@@ -206,7 +206,7 @@ public boolean createPatch(String oldPath, String newPath, String patchPath, Str
         }
 
         try (FileOutputStream fos = new FileOutputStream(patchPath);
-                ZipOutputStream zipOut = new ZipOutputStream(fos)) {
+             ZipOutputStream zipOut = new ZipOutputStream(fos)) {
```
Comment (pj892031) [inline jarpatcher/src/main/java/jarpatcher/JarPatcher.java:209]:
```suggestion
             BufferedOutputStream bos = new BufferedOutputStream(fos);
             ZipOutputStream zipOut = new ZipOutputStream(bos)
        ) {
```

+ import

File: jarpatcher/src/main/java/jarpatcher/JarPatcher.java [hunk 2]
```diff
@@ -235,31 +235,37 @@ public void applyPatch(String targetPath, String patchPath, String ignoredPath)
         String originalTargetPath = targetPath + "-original";
         Files.copy(Paths.get(targetPath), Paths.get(originalTargetPath), StandardCopyOption.REPLACE_EXISTING);
         try (FileOutputStream fos = new FileOutputStream(targetPath);
-                ZipOutputStream zipOut = new ZipOutputStream(fos)) {
+             ZipOutputStream zipOut = new ZipOutputStream(fos)) {
```
Comment (pj892031) [inline jarpatcher/src/main/java/jarpatcher/JarPatcher.java:238]:
```suggestion
             BufferedOutputStream bos = new BufferedOutputStream(fos);
             ZipOutputStream zipOut = new ZipOutputStream(bos)
      ) {
```

+ import

File: zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/apiml/ApimlIntegrationFailureDetector.java [hunk 3]
```diff
@@ -9,38 +9,33 @@
...
                 && (logger.getName().contains("DiscoveryClient")
-                        || logger.getName().contains("RedirectingEurekaHttpClient"))) {
+                || logger.getName().contains("RedirectingEurekaHttpClient"))) {
             if (logger.getLevel() == Level.ERROR) {
                 String message = ExceptionUtils.getMessage(t);
-                if (message == null) {
-                    message = ExceptionUtils.getRootCauseMessage(t);
-                }
-                if ((message != null) && !message.isEmpty()) {
+                if (!message.isEmpty()) {
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/apiml/ApimlIntegrationFailureDetector.java:38]: Are you sure, that getRootCauseMessage cannot return null? I guess it is better to use `!StringUtils.isEmpty(message)`

File: zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/error/ErrorServiceImpl.java [hunk 4]
```diff
@@ -235,8 +218,7 @@ private String getLocalizedComponentOrUseDefault(Locale locale, String key, Erro
         String component = message.getComponent();
         if (component == null) {
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/error/ErrorServiceImpl.java:219]:
```suggestion
        String component;
        if ( message.getComponent() == null) {
```

I guess it is better just check and set the right value, component could be final, it is better to optimize.

File: zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/error/ErrorServiceImpl.java [hunk 5]
```diff
@@ -235,8 +218,7 @@ private String getLocalizedComponentOrUseDefault(Locale locale, String key, Erro
...
             StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
-            String className = stackTrace[STACK_TRACE_ELEMENT_ABOVE_CREATEAPIMESSAGE_METHOD].getClassName();
-            component = className;
+            component = stackTrace[STACK_TRACE_ELEMENT_ABOVE_CREATEAPIMESSAGE_METHOD].getClassName();
         } else {
             component = localizedText(locale, key + ".component", component);
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/error/ErrorServiceImpl.java:223]:
```suggestion
            component = localizedText(locale, key + ".component", message.getComponent());
```

dtto above

File: zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/zos/LibExtractor.java [hunk 6]
```diff
@@ -30,16 +31,16 @@ public LibExtractionError(String message, IOException e) {
...
     public void extractLibrary(String libraryName, String targetDir) {
         String filename = "lib/lib" + libraryName + ".so";
         Path targetPath = Paths.get(targetDir, libLoader.libraryFileName(libraryName));
-        System.out.println(String.format("Extracting %s to %s", filename, targetPath));  // NOSONAR
+        System.out.printf("Extracting %s to %s%n", filename, targetPath);  // NOSONAR
         try {
             URL url = ResourceUtils.getURL(ResourceUtils.CLASSPATH_URL_PREFIX + filename);
             try (InputStream inputStream = url.openStream();
-                    OutputStream outputStream = new FileOutputStream(targetPath.toString())) {
+                 OutputStream outputStream = new FileOutputStream(targetPath.toString())) {
```
Comment (pj892031) [inline zowe-rest-api-commons-spring/src/main/java/org/zowe/commons/zos/LibExtractor.java:43]:
```suggestion
                 FileOutputStream fos = new FileOutputStream(targetPath.toString());
                 BufferedOutputStream outputStream = new BufferedOutputStream(fos)
        ) {
```
+ import

**Review by pj892031 — APPROVED** [review]

**Review by plavjanik — APPROVED** [review]

**Pull request merged**